standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -2174,6 +2174,22 @@ List<Partition> dropPartitions(String catName, String dbName, String tblName,
                                  PartitionDropOptions options)
       throws NoSuchObjectException, MetaException, TException;
 
+  /**
+   * @param catName catalog name
+   * @param dbName Name of the database
+   * @param tblName Name of the table
+   * @param partNames Arrays of partitions to drop
+   * @param options Boolean options for dropping partitions
+   * @return List of Partitions dropped
+   * @throws NoSuchObjectException No partition matches the expression(s), and ifExists was false.
+   * @throws MetaException error access the RDBMS or storage.
+   * @throws TException On failure
+   */
+  List<Partition> dropPartitions(String catName, String dbName, String tblName,
+                                 String[] partNames,
```

Comment:
Why do we want to declare this as String array? We are anyway using List<String> everywhere, right?

Reply:
Thanks @saihemanth-cloudera for the comment. Introduce `List<Partition> dropPartitions(DropPartitionsRequest req, PartitionDropOptions options)` instead, support HMS deleting partitions in batch by the provided partition names.